File: flutter_app/lib/theme/spacing.dart
import 'package:flutter/widgets.dart';

/// 应用间距常量
///
/// 统一管理应用中的间距值，保持一致性
class Spacing {
  Spacing._();

  static const double xs = 4.0;
  static const double sm = 8.0;
  static const double md = 16.0;
  static const double lg = 24.0;
  static const double xl = 32.0;
  static const double xxl = 48.0;
  static const double xxxl = 64.0;

  // EdgeInsets 预设
  static const allXS = EdgeInsets.all(xs);
  static const allSM = EdgeInsets.all(sm);
  static const allMD = EdgeInsets.all(md);
  static const allLG = EdgeInsets.all(lg);
  static const allXL = EdgeInsets.all(xl);

  static const horizontalMD = EdgeInsets.symmetric(horizontal: md);
  static const horizontalLG = EdgeInsets.symmetric(horizontal: lg);
  static const verticalMD = EdgeInsets.symmetric(vertical: md);
  static const verticalLG = EdgeInsets.symmetric(vertical: lg);

  static const onlyTopMD = EdgeInsets.only(top: md);
  static const onlyBottomMD = EdgeInsets.only(bottom: md);
}

File: rn_app/src/theme/spacing.ts
import { StyleSheet } from 'react-native';

/**
 * 应用间距常量
 * 统一管理应用中的间距值，保持一致性
 */
export const Spacing = {
  xs: 4,
  sm: 8,
  md: 16,
  lg: 24,
  xl: 32,
  xxl: 48,
  xxxl: 64,
} as const;

/**
 * EdgeInsets 预设样式
 */
export const SpacingStyles = StyleSheet.create({
  allXS: { padding: Spacing.xs },
  allSM: { padding: Spacing.sm },
  allMD: { padding: Spacing.md },
  allLG: { padding: Spacing.lg },
  allXL: { padding: Spacing.xl },

  horizontalMD: { paddingHorizontal: Spacing.md },
  horizontalLG: { paddingHorizontal: Spacing.lg },
  verticalMD: { paddingVertical: Spacing.md },
  verticalLG: { paddingVertical: Spacing.lg },

  onlyTopMD: { paddingTop: Spacing.md },
  onlyBottomMD: { paddingBottom: Spacing.md },

  // Margin presets
  marginBottomSM: { marginBottom: Spacing.sm },
  marginBottomMD: { marginBottom: Spacing.md },
  marginBottomLG: { marginBottom: Spacing.lg },
  marginTopMD: { marginTop: Spacing.md },
  marginTopLG: { marginTop: Spacing.lg },
});

export default Spacing;
